spot-ml/src/main/scala/org/apache/spot/utilities/data/InputOutputDataHandler.scala

```diff
@@ -74,4 +77,52 @@ object InputOutputDataHandler {
       s"Please check previous logs to see if there were errors.")
   }
 
+  def mergeResultsFileSystem(sparkSession: SparkSession,
+                             hdfsScoredConnect: String,
+                             analysis: String,
+                             logger: Logger): Unit = {
+
+    val hadoopConfiguration = sparkSession.sparkContext.hadoopConfiguration
+    val fileSystem = org.apache.hadoop.fs.FileSystem.get(hadoopConfiguration)
+
+    val exists = fileSystem.exists(new org.apache.hadoop.fs.Path(hdfsScoredConnect))
+
+    if (exists) {
+      // SPOT-172. Seems like FilUtil.copyMerge is not behaving well with Proxy data. Besides, it's deprecated after
+      // Hadoop 2.7.x
+      // Using spark to merge result files
+      val tmpFileStr = s"${hdfsScoredConnect}/tmp"
+      val tmpPath = new Path(s"${hdfsScoredConnect}/tmp")
+
+      // File name after sparkContext.textFile.coalesce(1)
+      val singleFileName = "part-00000"
+
+      val srcPath = new Path(hdfsScoredConnect)
+      val srcPartsStr = s"${hdfsScoredConnect}/part-*"
+
+      val srcTmpPath = new Path(s"${tmpFileStr}/${singleFileName}")
+      val dstTmpPath = new Path(s"${hdfsScoredConnect}/${analysis}_results.csv")
+
+      // Merge all part-* into a single file
+      sparkSession.sparkContext.textFile(srcPartsStr).coalesce(1).saveAsTextFile(tmpFileStr)
+
+      // Rename the single file generated part-00000 to ${analysis}_results.csv
```

Comment:
why bother with the intermediate name part-00000?
couldn't we just go straight to ${analysis}_results.csv ?